metric/meter.go

```diff
@@ -116,7 +118,15 @@ type Meter interface {
 // the same attributes as another Callback will report.
 //
 // The function needs to be concurrent safe.
-type Callback func(context.Context) error
+type Callback func(context.Context, MultiObserver) error
+
+// MultiObserver records measurements for multiple instruments in a Callback.
+type MultiObserver interface {
```

Comment:
How about Observer instead of MultiObserver?  I have to write this each time I make a callback, so shorter seems nice.  Is there already an Observer type this would be confused with?  Other alternative if observer is overloaded: Recorder.

Reply:
The reason I went with `MultiObserver` here is to parallel the `Float64Observer` and `Int64Observer` from the `instrumentation` package. I was concerned that naming this `Observer` and then having `Float64Observer` and `Int64Observer` in addition it wouldn't be clear from the name what distinguished these types.

That said, your point on use is very relevant. I'm not entirely sure of an explicit example when a user will use an `Observer` and be confused `Float64Observer` and `Int64Observer` and `Observer` is shorter than `MultiObserver`.

@dashpole given my above concern are you still in favor of renaming? I'm good with it if you are.

Reply:
FWIW, updated with rename. LMK if you have thoughts otherwise.

Reply:
Rename looks good.  Can you update the description?

Reply:
Updated.